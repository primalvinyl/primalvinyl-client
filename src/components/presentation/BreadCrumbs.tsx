import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import styles from '../../styles/partials/_variables.module.scss';

const Wrapper = styled.nav`
    margin: 0;
    padding: .5em 1em;
    color: ${styles.gray600};
    background: ${styles.gray100};
    border-radius: 5px;

    a {
        color: ${styles.gray600};
        text-decoration: none;

        &:hover {
            text-decoration: underline;
        }
    }
`;

const BreakCrumbs = ({ currentPage, previousPages }: BreakCrumbsProps): React.ReactElement => {
    const breadCrumbList = previousPages.map(listItem => (
        <span key={listItem.link}>
            <Link to={listItem.link}>{listItem.title}</Link>
            {currentPage !== '' &&
                <span>&nbsp;&nbsp;&gt;&nbsp;&nbsp;</span>
            }
        </span>
    ));
    
    return (
        <Wrapper>
            {currentPage === '' &&
                <span>&lt;&nbsp;&nbsp;</span>
            }
            {breadCrumbList}
            {currentPage !== '' &&
                <span>{currentPage}</span>
            }
        </Wrapper>
    );
};

type previousPagesItem = {
    title: string;
    link: string;
};

type BreakCrumbsProps = {
    currentPage?: string;
    previousPages: previousPagesItem[];
};

BreakCrumbs.defaultProps = {
    currentPage: '',
    previousPages: [{ title: '', link: '' }]
};

export default BreakCrumbs;
