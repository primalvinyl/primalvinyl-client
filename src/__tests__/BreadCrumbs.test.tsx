import * as React from 'react';
import { shallow } from 'enzyme';
import Renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import BreadCrumbs from '../components/presentation/BreadCrumbs';

describe('BreadCrumbs', () => {
    const previousPages = [
        { title: 'test', link: 'test' }
    ];

    it('renders', () => {
        const myWrapper = shallow(
            <MemoryRouter>
                <BreadCrumbs
                    currentPage="test"
                    previousPages={previousPages} />
            </MemoryRouter>
        );
        expect(myWrapper.exists());
    });

    it('matches snapshot', () => {
        const component = Renderer.create(
            <MemoryRouter>
                <BreadCrumbs
                    currentPage="test"
                    previousPages={previousPages} />
            </MemoryRouter>
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});