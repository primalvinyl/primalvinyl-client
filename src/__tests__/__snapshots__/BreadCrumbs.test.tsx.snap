// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs matches snapshot 1`] = `
<nav
  className="sc-bdfBwQ hKuajV"
>
  <span>
    <a
      href="/test"
      onClick={[Function]}
    >
      test
    </a>
    <span>
        &gt;  
    </span>
  </span>
  <span>
    test
  </span>
</nav>
`;
